src/SMU-Agilent_B29xx/main.py

```diff
@@ -390,16 +399,43 @@ def apply(self):
         value = str("{:.4E}".format(self.value))  # makes sure that self.value fits into SCPI command in terms of length
         if self.pulse:
             # get channel ready at specified values
+ 
             self.port.write(":SOUR%s:%s %s" % (self.channel, self.commands[self.source], self.pulseofflevel))
-            # arming the pulse trigger
-            self.port.write(":SOUR%s:%s:TRIG %s" % (self.channel, self.commands[self.source], value))
+    
+            # arming the pulse trigger; mis-using the sweep function with identical start and stop values so we can use number-of-pulses functionality
+            if self.source.startswith("Current"):
+                self.port.write(":SOUR%s:CURR:STAR %s" % (self.channel, value))
+                self.port.write(":SOUR%s:CURR:STOP %s" % (self.channel, value))
+            
+            if self.source.startswith("Voltage"):
+                self.port.write(":SOUR%s:VOLT:STAR %s" % (self.channel, value))
+                self.port.write(":SOUR%s:VOLT:STOP %s" % (self.channel, value))
+            
             # releasing the pulse trigger
             self.port.write(":INIT (@%s)" % self.channel)
```

Comment:
Releasing the pulse could be also shifted to a later phase. This can be beneficial when for example the measurement must be synchronized with other instruments. For example the Keithley 26xx driver is initiating the pulse in the "trigger_ready" phase just before "measure" phase.

Example: A motor and the pulsed SMU get a new set value during "apply". The motor needs some time to reach the new position. This is checked in the "reach" phase after all instruments had their "apply" phase. However, the SMU already sent the pulse during "apply". It would be not a problem in a DC measurement, but for pulsed measurements it is more important that the pulse is triggered after "reach" and "adapt" phase, just before the "measure".

For measurements with only one pulsed SMU there should be however no problem.